go/proxyd/daisychain.go

```diff
@@ -0,0 +1,528 @@
+package proxyd
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"reflect"
+	"strings"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/gorilla/mux"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/rs/cors"
+)
+
+// TODO: should get by hash work its way backwards
+// until it finds the value?
+
+type DaisyChainServer struct {
+	rpcServer    *http.Server
+	wsServer     *http.Server
+	maxBodySize  int64
+	epoch1RPCURL string
+	epoch2RPCURL string
+	epoch3RPCURL string
+	epoch4RPCURL string
+	epoch5RPCURL string
+	epoch6RPCURL string
+	client       *http.Client
+}
+
+// TODO: support "latest" for epoch
+type RequestOptions struct {
+	Epoch *uint `json:"epoch,omitempty"`
+}
+
+var latestEpoch = uint(6)
+
+// TODO: also add in debug methods
+var argTypes = map[string][]reflect.Type{
+	"eth_blockNumber": []reflect.Type{
```

Comment:
Usually not a fan of `reflect` but since this is global it shouldn't affect performance much. 👍🏽